packages/prisma/zod-utils.ts

```diff
@@ -13,6 +13,7 @@ import type {
   ZodTypeAny,
 } from "zod";
 
+import { timeZone } from "@calcom/api/lib/validations/shared/timeZone";
```

Comment:
I don't think we use tzdata package which is inside /shared/timeZone file. we use react-timezone-select to get the lost of available timezones

Reply:
I also had second thoughts about using that package. Here's an alternate version without using list of all timezones.

```typescript
  timeZone: z.string().refine(
    (value: string) => {
      try {
        dayjs().tz(value);
        return true;
      } catch (error) {
        return false;
      }
    },
    { message: "Invalid timezone" }
  )
```

![image](https://github.com/calcom/cal.com/assets/8333321/c0ea75b6-2477-4eda-9bad-fe775af39061)

Reply:
@jaypitroda12 we use react-timezone-select and city-timezones in TimezoneSelect.tsx. I am not sure if  dayjs().tz(value) would work exactly same as them

Reply:
@Udit-takkar day.js uses has IANA database for timezones [ref](https://day.js.org/docs/en/timezone/timezone)

both `react-timezone-select` and `city-timezones` has different set of timezones from what I've checked, so maybe we can use `city-timezones` in this case for validation?

Reply:
@Udit-takkar I think it is best if we take consider the set we use in parsing the timezone when creating a booking in the backend, or in in the `cal.com/bookings` page instead, no? The issue was highlighted by API users alone so far. Plus, we can potentially adjust react-timezone-select to only include the supported timezones which should arguably be the easier thing to do 🤔

Reply:
@alishaz-polymath thank you for taking over the critical part. 🙌🏼 

I've updated the PR, is there something more to be covered for the issue?